DEV/org.openl.rules.util/src/org/openl/rules/util/Arrays.java

```diff
@@ -266,4 +239,140 @@ private static Object subarray(Object array, int beginIndex, int endIndex) {
     private static <T> T[] clone(final T[] array) {
         return array == null ? null : array.clone();
     }
+
+    /**
+     * Removes the first occurrence of the specified element from the specified array. All subsequent elements are
+     * shifted to the left (substracts one from their indices). If the array does not contains such an element, no
+     * elements are removed from the array. <br />
+     * <br />
+     * <p/>
+     * This method returns a new array with the same elements of the input array except the first occurrence of the
+     * specified element. The component type of the returned array is always the same as that of the input array. <br />
+     * <br />
+     * <p/>
+     * <code>
+     * ArrayUtils.removeElement(null, "a")            = null        <br />
+     * ArrayUtils.removeElement([], "a")              = []          <br />
+     * ArrayUtils.removeElement(["a"], "b")           = ["a"]       <br />
+     * ArrayUtils.removeElement(["a", "b"], "a")      = ["b"]       <br />
+     * ArrayUtils.removeElement(["a", "b", "a"], "a") = ["b", "a"]  <br />
+     * </code>
+     *
+     * @param array the array to remove the element from, may be null
+     * @param elements
+     * @return the element to be removed
+     */
+    public static <T> T[] removeElement(T[] array, T... elements) {
+        if (array == null) {
+            return null;
+        }
+        if (elements == null) {
+            return array;
+        }
+        ArrayList<T> result = new ArrayList<>(array.length);
+        result.addAll(java.util.Arrays.asList(array));
+         java.util.Arrays.stream(elements).forEach(result::remove);
+        return (T[]) result.toArray();
+    }
+
+    /**
+     * Return a new array without null elements
+     *
+     * @param elements whose null elements should be removed
+     * @return new array without null elements
+     */
+
+    public static <T> T[] removeNulls(T... elements) {
+        if (elements == null) {
+            return null;
+        }
+        T[] result;
+
+        int valuableSize = getNotNullValuesCount(elements);
+        if (elements == null || valuableSize == elements.length) {
+            result = elements;
+        } else {
+            result = (T[]) Array.newInstance(elements.getClass().getComponentType(), valuableSize);
+            int i = 0;
+            for (T value : elements) {
+                if (value != null) {
+                    result[i] = value;
+                    i++;
+                }
+            }

```

Comment:
This logic can be improved:

- if ``valuableSize == 0`` is true, then return empty array
- if ``i == valuableSize`` then break iteration, because resulted array is full